llvm/lib/ProfileData/RawMemProfReader.cpp

```diff
@@ -445,14 +448,20 @@ Error RawMemProfReader::mapRawProfileToRecords() {
       Callstack.append(Frames.begin(), Frames.end());
     }
 
+    auto InsertResult =
+        CallStackToCallStackId.insert({Callstack, CallStacks.size()});
```

Comment:
As discussed offline, it might be better to use a content hash (I.e. hash of the frame IDs) as the call stack ID. You can refactor and reuse the frame ID hashing implementation here [1]  which is suitable for IDs meant to be persisted to disk. Factoring it out would also mean we can reuse it for @teresajohnson's use case of tagging callstacks with statistics. 

[1] https://github.com/llvm/llvm-project/blob/84115494d6475e1aea3cdd1163d3a88243b75f36/llvm/include/llvm/ProfileData/MemProf.h#L237-L253

Reply:
I just switched to the content hash for call stacks in the latest iteration.  Since computing a hash value requires no data structure to speak of (unlike the sequential `CallStackId`), I've updated the tests to call `hashCallStack` instead of the placeholder `0`.

For now, I am using the same hash function as in `computeStackId` in `MemProfiler.cpp`, but I can switch to `std::hash` also.